packages/fx-core/src/plugins/solution/fx-solution/v2/utils.ts

```diff
@@ -196,8 +198,13 @@ export function fillInSolutionSettings(
     );
   }
   let hostType = answers[AzureSolutionQuestionNames.HostType] as string;
-  if (capabilities.includes(BotOptionItem.id) || capabilities.includes(MessageExtensionItem.id))
+  if (
+    capabilities.includes(BotOptionItem.id) ||
+    capabilities.includes(MessageExtensionItem.id) ||
+    capabilities.includes(TabOptionItem.id)
+  )
     hostType = HostTypeOptionAzure.id;
+  if (capabilities.includes(TabSPFxItem.id)) hostType = HostTypeOptionSPFx.id;
```

Comment:
I think there's much more than here that you need to update since currently there're many places use capabilities.includes(TabOptionItem.id) to indicate whether there's tab cap.

Reply:
fixed.